src/main/java/com/microsoft/sqlserver/jdbc/SQLServerResultSet.java

```diff
@@ -675,14 +685,15 @@ public int findColumn(String columnName) throws SQLServerException {
         // Per JDBC spec, 27.3 "The driver will do a case-insensitive search for
         // columnName in it's attempt to map it to the column's index".
         // Use VM supplied String.equalsIgnoreCase to do the "case-insensitive search".
-        for (i = 0; i < columns.length; i++) {
-            if (columns[i].getColumnName().equalsIgnoreCase(columnName)) {
+        for (int i = 0; i < columns.length; i++) {
+            if (columns[i].getColumnName().equalsIgnoreCase(userProvidedColumnName)) {
```

Comment:
Do we use `equalsIgnoreCase` or `TreeMap` 🤔 
Use one approach in both classes.

Reply:
This is separate from the treemap, we have to do it differently in Resultset/CallableStatement because one's an array of columns and the other just has some names.

Reply:
The String in `columnNames` HashMap contains names of column in both implementations, its just that here its fetched from column Array. The purpose of Map is same.

Reply:
You can't do a String insensitive look up on a hashmap Key.